src/modal/modal-window.ts

```diff
@@ -76,4 +82,15 @@ export class NgbModalWindow implements OnInit,
     this._elWithFocus = null;
     this._renderer.setElementClass(document.body, 'modal-open', false);
   }
+
+  private _setAriaLabelledByToModal(): void {
+    const modalHeader: HTMLHeadingElement = this._elRef.nativeElement.getElementsByClassName('modal-title')[0];
+    if (modalHeader) {
+      const modalHeaderId: string = modalHeader.getAttribute('id') === null || modalHeader.getAttribute('id') === '' ?
+          `modal-header-${nextId++}` :
+          modalHeader.getAttribute('id');
+      this._renderer.setElementAttribute(modalHeader, 'id', modalHeaderId);
+      this._renderer.setElementAttribute(this._elRef.nativeElement, 'aria-labelledby', modalHeaderId);
+    }
+  }
```

Comment:
If one wanted to do direct DOM manipulation like this, it would be better done on the user side by manually walking up the DOM tree until it reached the parent to set the id via the renderer.

Another option is to add a host binding for `[attr.aria-labelledby]` on the modal window component, and let the user set the id on the window component instance on the NgbModalRef that is returned from `modal.open`. This would also let us autogenerate an id in the component that the user can read for use in binding on the title...which also opens up the potential for us adding a lightweight NgbModalTitleDirective whose purpose is to add the id attribute to the element for use with the aria labelled-by attribute.